### ISSUE TITLE
[INTEGRATION][Spark] Cover Command in org.apache.spark.sql.execution.command

### ISSUE THREAD

**Issue by mobuchowski** [OpenLineage/OpenLineage#371]
Cover Command in 
            org.apache.spark.sql.execution.command
                    CreateDataSourceTableAsSelectCommand
                    DropTableCommand
                    AlterTableChangeColumnCommand
                    AlterTableAddPartitionCommand
                    AlterTableRenamePartitionCommand
                    AlterTableDropPartitionCommand
                    AlterTableRecoverPartitionsCommand
                    InsertIntoDataSourceDirCommand
                    CreateTableCommand
                    AlterTableRenameCommand
                    AlterTableAddColumnsCommand
                    LoadDataCommand
                    TruncateTableCommand

**Comment by wjohnson** [comment]
@mobuchowski our team wanted to attempt an "easy" one like this issue (with only three commands)

However, it looks like this may have already been covered?

* [InsertIntoDataSourceVisitor](https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/InsertIntoDataSourceVisitor.java)
* [InsertIntoHadoopFsRelationVisitor](https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/InsertIntoHadoopFsRelationVisitor.java)
* [SaveIntoDataSourceVisitor](https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/SaveIntoDataSourceCommandVisitor.java)

Is this issue still really open and if so, can you share more about where you need support so we can attempt to contribute?

**Comment by mobuchowski** [comment]
@wjohnson yes, those are covered - keeping in mind that SaveIntoDataSource covers quite a bit of different relations, so some use cases can still be missing.

I think we're mostly left with those: https://github.com/OpenLineage/OpenLineage/issues/368
Those ones are DataFrameWriterV2 interfaces. The caveat with them is that they might have implementation dependent on "provider" you're using them with.
CreateTableAsSelect is done in this PR for Iceberg and Delta, as well as utilities for handling v2 catalogs - those might be helpful: https://github.com/OpenLineage/OpenLineage/pull/419

Feel free to take some (or even one) of those.
